limbus/core/component.py

```diff
@@ -125,12 +125,6 @@ def __init__(self, name: str):
         # Last execution to be run in the __call__ loop.
         self._stopping_iteration: int = 0  # 0 means run forever
 
-        # method called in _run_with_hooks to execute the component forward method
-        self._run_forward: Callable[..., Coroutine[Any, Any, ComponentState]] = self.forward
-        if nn.Module in Component.__mro__:
-            # If the component inherits from nn.Module, the forward method is called by the __call__ method
-            self._run_forward = nn.Module.__call__
-
```

Comment:
I got PyTorch 1.13 here. I do not think it is a pytorch issue, since it complains regarding the "self" parameter.

```
Traceback (most recent call last):
  File "C:\Users\SHIJ0C\git\limbus\limbus\core\component.py", line 335, in _run_with_hooks
    self.set_state(await self._run_forward(*args, **kwargs))
TypeError: _call_impl() missing 1 required positional argument: 'self'

INFO:limbus.core.component: printer0(1): STOPPED_AT_ITER
INFO:limbus.core.component: add(1): READY
ERROR:limbus.core.component:Error in component add.
Traceback (most recent call last):
  File "C:\Users\SHIJ0C\git\limbus\limbus\core\component.py", line 335, in _run_with_hooks
    self.set_state(await self._run_forward(*args, **kwargs))
TypeError: _call_impl() missing 1 required positional argument: 'self'

INFO:limbus.core.component: add(1): STOPPED_AT_ITER
INFO:limbus.core.pipeline: ENDED
```

Reply:
I could reproduce the error the error!!! I will check it now.

Reply:
Problem address, I forgot to pass the `self` param. The correct code there is:
```python
        try:
            if nn.Module in Component.__mro__:
                # If the component inherits from nn.Module, the forward method is called by the __call__ method
                self._run_forward = partial(nn.Module.__call__, self)
        except NameError:
            pass
```

remember to add `from functools import partial` at the top :)

If you want i can add the fix. Just let me know.